kernel/src/actions/statistics.rs

```diff
@@ -0,0 +1,272 @@
+//! This module defines the `Statistics` struct used in Add and Remove actions.
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Statistics {
```

Comment:
I implemented `Statistics` here because:
- They are defined in the context of `Add` actions.
- Having them as a public interface lets engines reuse them out of the box.

Currently, the struct is only used in the default engine so strictly speaking, we could move it there. IMO, the main decision point is whether we want to provide this as a "service" the other engines or not.